### PR TITLE
Add optional String function to allow passing nil Strings

### PR DESCRIPTION
**Pull request by ShihabMehboob**
Added an extra routerResponse function that takes in optional Strings to allow nil Strings to be sent without resorting to matching the second unlabelled JSON object function. In relation to this issue: https://github.com/IBM-Swift/Kitura/issues/1307

### CODE REVIEW

File: Sources/Kitura/RouterResponse.swift [hunk 1]
```diff
@@ -222,6 +222,28 @@ public class RouterResponse {
...
+    ///
+    /// - Parameter str: the string to send.
+    /// - Returns: this RouterResponse.
+    @discardableResult
+    public func send(_ str: String?) -> RouterResponse {
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(str:) invoked after end() for \(self.request.urlURL)")
+            return self
+        }
+        var stringToSend: String
```
Comment (ianpartridge) [inline Sources/Kitura/RouterResponse.swift]: Couldn't this whole thing just be:

```swift
guard let str = str else {
    Log.warning("RouterResponse send(str:) invoked with nil")
    return send("")
}
return send(str)
```

File: .swift-version [hunk 2]
```diff
@@ -1 +1 @@
-4.1.2
+4.1
```
Comment (ianpartridge) [inline .swift-version]: ?

File: Tests/KituraTests/TestResponse.swift [hunk 3]
```diff
@@ -1493,6 +1493,8 @@ class TestResponse: KituraTest {
 
                 try response.send(status: HTTPStatusCode.forbidden).send(data: "<!DOCTYPE html><html><body><b>forbidden</b></body></html>\n\n".data(using: .utf8)!).end()
                 try response.send(status: HTTPStatusCode.OK).end()
+                let emptyString: String? = nil
```
Comment (ianpartridge) [inline Tests/KituraTests/TestResponse.swift]: This should be named `nilString`.

File: Sources/Kitura/RouterResponse.swift [hunk 4]
```diff
@@ -222,6 +222,19 @@ public class RouterResponse {
...
     }
+    
+    /// Send an optional string.
+    ///
+    /// - Parameter str: the string to send.
+    /// - Returns: this RouterResponse.
+    @discardableResult
+    public func send(_ str: String?) -> RouterResponse {
+        guard !state.invokedEnd, let str = str else {
+            Log.warning("RouterResponse send(str:) invoked after end() for \(self.request.urlURL)")
```
Comment (djones6) [inline Sources/Kitura/RouterResponse.swift]: This doesn't look right, you've rolled two different checks into one guard, when they are checking two distinct cases:
- `state.invokedEnd` should log a warning ("send invoked after end") and return `self`,
- `str == nil` should log a different warning ("send invoked with a nil string") and return `send("")`.

**Comment by ShihabMehboob** [comment]
@ianpartridge Thanks, made those test changes now to be more clear.

File: Tests/KituraTests/TestResponse.swift [hunk 5]
```diff
@@ -101,6 +103,38 @@ class TestResponse: KituraTest {
...
     }
+    
+    func testOptionalStringNilResponse() {
+        performServerTest(router) { expectation in
+            self.performRequest("post", path:"/sendNilString", callback: { response in
+                XCTAssertEqual(response?.statusCode, HTTPStatusCode.OK, "HTTP Status code was \(String(describing: response?.statusCode))")
+                
+                do {
+                    let body = try response?.readString()
+                    XCTAssertNil(body)
```
Comment (ianpartridge) [inline Tests/KituraTests/TestResponse.swift]: I thought in the nil case we were sending the empty string - so wouldn't we expect to get an empty body back?

**Comment by ShihabMehboob** [comment]
@ianpartridge Realised that the default readString() function would always return nil if the String was empty, opened a PR in Kitura-Net to amend this: https://github.com/IBM-Swift/Kitura-net/pull/274

**Comment by ShihabMehboob** [comment]
@ianpartridge Sure, amended the test to use readAllData() instead, removing the reliance on the new Kitura-net API.

**Pull request merged**